designsafe/apps/api/publications/models.py

```diff
@@ -0,0 +1,27 @@
+"""
+Django model for storing symlink metadata for published Tapis files.
+This table is used to resolve the actual type (file or dir) of symbolic links
+in the published data system, since Tapis cannot always determine the target type.
+"""
+
+from django.db import models
+
+
+class PublicationSymlink(models.Model):
```

Comment:
Let's move this model to the `datafiles` app and rename it to `DatafilesSymlinkType`, since it's a generic way to handle symlinks that we might use for other things in the future.

This diff should also have the migration for the model (`python manage.py makemigrations datafiles`)